js/dialogcards.js

```diff
@@ -481,6 +481,13 @@ H5P.Dialogcards = (function ($, Audio, JoubelUI) {
     // Check if card has been turned before
     var turned = $c.hasClass('h5p-dialogcards-turned');
 
+	//If the checkbox to use the same front image on back also is checked, show the front image only regardless of whether a back image is uploaded or not.
+	//P.S. Adding this snippet after changeText() caused image sizing problems
+	if(self.params.dialogs[$card.index()]['use_same'])
```

Comment:
There should be no need to check for `use_same` option. Check if card has a `back_image`, then toggle between `back_image` and `image` depending on whether the card is turned.

Also `self.params.dialogs[$card.index()]` is an object, so you can access any attributes with `self.params.dialogs[$card.index()].image`, which is the preferred coding style.